.mergify.yml

```
@@ -106,31 +116,49 @@ pull_request_rules:
-      - "status-success=codespell"
-      - "status-success=multi-arch-build"
-      - "status-success=go-test"
-      - "status-success=golangci-lint"
-      - "status-success=mod-check"
-      - "status-success=lint-extras"
-      - "status-success=ci/centos/k8s-e2e-external-storage/1.30"
-      - "status-success=ci/centos/k8s-e2e-external-storage/1.31"
-      - "status-success=ci/centos/k8s-e2e-external-storage/1.32"
-      - "status-success=ci/centos/mini-e2e-helm/k8s-1.30"
-      - "status-success=ci/centos/mini-e2e-helm/k8s-1.31"
-      - "status-success=ci/centos/mini-e2e-helm/k8s-1.32"
-      - "status-success=ci/centos/mini-e2e/k8s-1.30"
-      - "status-success=ci/centos/mini-e2e/k8s-1.31"
-      - "status-success=ci/centos/mini-e2e/k8s-1.32"
-      - "status-success=ci/centos/upgrade-tests-cephfs"
-      - "status-success=ci/centos/upgrade-tests-rbd"
-      - "status-success=DCO"
+      - or:
+          - and:
+              - label=ci/skip/e2e
+              - author=dependabot[bot]
+              - label!=DNM
+              - base=devel
+              - "#approved-reviews-by>=2"
+              - "#changes-requested-reviews-by=0"
+              - "approved-reviews-by=@ceph/ceph-csi-contributors"
+              - "approved-reviews-by=@ceph/ceph-csi-maintainers"
+              - "status-success=codespell"
+              - "status-success=multi-arch-build"
+              - "status-success=go-test"
+              - "status-success=golangci-lint"
+              - "status-success=mod-check"
+              - "status-success=lint-extras"
+              - "status-success=DCO"
+          - and:
+              - author=dependabot[bot]
+              - label!=DNM
+              - base=devel
+              - "#approved-reviews-by>=2"
+              - "#changes-requested-reviews-by=0"
+              - "approved-reviews-by=@ceph/ceph-csi-contributors"
+              - "approved-reviews-by=@ceph/ceph-csi-maintainers"
+              - "status-success=codespell"
+              - "status-success=multi-arch-build"
+              - "status-success=go-test"
+              - "status-success=golangci-lint"
+              - "status-success=mod-check"
+              - "status-success=lint-extras"
+              - "status-success=ci/centos/k8s-e2e-external-storage/1.30"
+              - "status-success=ci/centos/k8s-e2e-external-storage/1.31"
+              - "status-success=ci/centos/k8s-e2e-external-storage/1.32"
+              - "status-success=ci/centos/mini-e2e-helm/k8s-1.30"
+              - "status-success=ci/centos/mini-e2e-helm/k8s-1.31"
+              - "status-success=ci/centos/mini-e2e-helm/k8s-1.32"
+              - "status-success=ci/centos/mini-e2e/k8s-1.30"
+              - "status-success=ci/centos/mini-e2e/k8s-1.31"
+              - "status-success=ci/centos/mini-e2e/k8s-1.32"
+              - "status-success=ci/centos/upgrade-tests-cephfs"
+              - "status-success=ci/centos/upgrade-tests-rbd"
```

```suggestion
             -or:
              - label=ci/skip/e2e
              - and:
                - "status-success=ci/centos/k8s-e2e-external-storage/1.30"
                - "status-success=ci/centos/k8s-e2e-external-storage/1.31"
                - "status-success=ci/centos/k8s-e2e-external-storage/1.32"
                - "status-success=ci/centos/mini-e2e-helm/k8s-1.30"
                - "status-success=ci/centos/mini-e2e-helm/k8s-1.31"
                - "status-success=ci/centos/mini-e2e-helm/k8s-1.32"
                - "status-success=ci/centos/mini-e2e/k8s-1.30"
                - "status-success=ci/centos/mini-e2e/k8s-1.31"
                - "status-success=ci/centos/mini-e2e/k8s-1.32"
                - "status-success=ci/centos/upgrade-tests-cephfs"
                - "status-success=ci/centos/upgrade-tests-rbd"
```
I think something like this is better ?

---

Can you adopt this same format at all places where centos-ci status is being checked ? 

release branches are being left out currently in this pr.

---

Adapted this format for `pull_request_rules` and `queue_rules`.